### PR TITLE
docs wrt using accelerate launcher with trainer

### PR DESCRIPTION
**Pull request by pacman100**
# What does this PR do?
1. Major issue currently is the confusion about how to use accelerate launcher with Trainer. This PR addresses it.

### CODE REVIEW

File: docs/source/en/main_classes/trainer.mdx [hunk 1]
```diff
@@ -688,6 +688,160 @@ Finally, please, remember that, 🤗 `Trainer` only integrates MPS backend, ther
...
+- They can keep using the Trainer ingterations such as FSDP, DeepSpeed vis trainer arguments without any changes on their part.
+- They can now use Accelerate Launcher with Trainer giving them following benefits:
+  - Use the same `accelerate` config between scripts that either leverage Trainer or are using Accelerate with custom loops.
+  - We won't be adding new training arguments for the Trainer Integrations.
+    They must be aupported in Accelerate as it trickles down to Trainer. In such cases,
+    the new paramters or updates require usage of Accelerate Launcher.
+  - New features and integrations in Accelerate can be leveraged out-of-the-box in Trainer via Accelerate Launcher/Config
+
+Steps to use Accelerate Launcher with Trainer:
+1. Make sure 🤗 Accelerate is installed. If not `pip install accelerate`
```
Comment (sgugger) [inline docs/source/en/main_classes/trainer.mdx:704]:
```suggestion
1. Make sure 🤗 Accelerate is installed, you can't use the `Trainer` without it anyway. If not `pip install accelerate`. You may also need to update your version of Accelerate: `pip install accelerate --upgrade`
```

File: docs/source/en/main_classes/trainer.mdx [hunk 2]
```diff
@@ -688,6 +688,160 @@ Finally, please, remember that, 🤗 `Trainer` only integrates MPS backend, ther
...
+2. Run `accelerate config --config_file xyz.yaml` and fill the questionnaire. Below are example accelerate configs:
```
Comment (sgugger) [inline docs/source/en/main_classes/trainer.mdx:705]:
```suggestion
2. Run `accelerate config` and fill the questionnaire. Below are example accelerate configs:
```
No need to trouble the user with a custom config file.

File: docs/source/en/main_classes/trainer.mdx [hunk 3]
```diff
@@ -688,6 +688,160 @@ Finally, please, remember that, 🤗 `Trainer` only integrates MPS backend, ther
...
+    same_network: true
+    tpu_env: []
+    tpu_use_cluster: false
+    tpu_use_sudo: false
+    use_cpu: false
+    ```
+
+3. Run the Trainer script with args other than the ones handled above by accelerate config or launcher args.
+Below is an example to run `run_glue.py` using `accelerate launcher` with FSDP config from above. 
+Provide the appropriate `config_file`.
```
Comment (sgugger) [inline docs/source/en/main_classes/trainer.mdx:801]: Providing the config_file should be in a more advanced section.

**Review by sgugger — COMMENTED** [review]
Thanks for drafting this, left a couple of comments.

**Review by muellerzr — APPROVED** [review]
Thanks! Let's also leave a link to the CLI guide as well please: https://huggingface.co/docs/accelerate/basic_tutorials/launch

File: docs/source/en/main_classes/trainer.mdx (hunk 3 above)
Comment (pacman100) [reply docs/source/en/main_classes/trainer.mdx]: done.

File: docs/source/en/main_classes/trainer.mdx [hunk 4]
```diff
@@ -688,6 +688,160 @@ Finally, please, remember that, 🤗 `Trainer` only integrates MPS backend, ther
...
+## Using Accelerate Launcher with Trainer
+
+Accelerate now powers Trainer. In terms of what users should expect:
+- They can keep using the Trainer ingterations such as FSDP, DeepSpeed vis trainer arguments without any changes on their part.
+- They can now use Accelerate Launcher with Trainer giving them following benefits:
+  - Use the same `accelerate` config between scripts that either leverage Trainer or are using Accelerate with custom loops.
+  - We won't be adding new training arguments for the Trainer Integrations.
+    They must be aupported in Accelerate as it trickles down to Trainer. In such cases,
+    the new paramters or updates require usage of Accelerate Launcher.
+  - New features and integrations in Accelerate can be leveraged out-of-the-box in Trainer via Accelerate Launcher/Config
```
[the remark this answers is not in the document]
Comment (pacman100) [reply docs/source/en/main_classes/trainer.mdx]: done.

**Review by sgugger — APPROVED** [review]

**Pull request merged**